### DISCUSSION TITLE
Version 2 pre-release / release master ticket

### DISCUSSION THREAD

**Discussion by vincentsarago** [cogeotiff/rio-tiler#163] [General]
## 2.0a.1 - 2020/03/19

I believe #154 is in a good place right now, so I'll cut an alpha release. 

This is a MAJOR release and will break everything, I've started a doc explaining most of the important change and how to switch from version 1.3.1 to 2.0 https://github.com/cogeotiff/rio-tiler/blob/v2/docs/v2_migration.md

I'll update couple rio-tiler based projects (rio-viz, rio-tiler-mosaic) when the alpha is in pypi

**Comment by jqtrde** [comment]
![](https://66.media.tumblr.com/tumblr_ma0do9CrEh1qi2sux.gif)

**Comment by vincentsarago** [comment]
🎉 https://pypi.org/project/rio-tiler/2.0a1/

**Comment by vincentsarago** [comment]
## 2.0a.2 - 2020/03/20

I've added a fix to #164 and created a new alpha release which should be up in pypi now

**Comment by vincentsarago** [comment]
## 2.0a.3 - 2020/03/25

fix #164 and #166

**Comment by vincentsarago** [comment]
## 2.0a.4 - 2020/04/08
https://github.com/cogeotiff/rio-tiler/blob/eacf16acd109e76bba92380c2d2096b75528d290/CHANGES.txt#L2-L6

This version adds a simple `cogeo.info` method to return non image statistic metadata and a ìo.stac` submodule to make handling STAC item easier.

**Comment by vincentsarago** [comment]
## 2.0a5 - 2020/05/06

https://github.com/cogeotiff/rio-tiler/blob/08e352a1539a8bfbedd07382c11bacfdcace787d/CHANGES.txt#L1-L5

**Comment by vincentsarago** [comment]
## 2.0a6 - 2020/05/06
Sorry about the breaking change introduced in 2.0a5, it should be fixed by https://github.com/cogeotiff/rio-tiler/pull/180 and released on pypi as I write this

**Comment by vincentsarago** [comment]
FYI, to ease the transition and ongoing development toward a release of Version 2 we are going to merge the `v2` branch in Master

**Comment by vincentsarago** [comment]
## 2.0a7 - 2020/05/17

https://github.com/cogeotiff/rio-tiler/blob/aa07d976f0ba67f754776c2af662e766b29506d6/CHANGES.txt#L1-L9

**Comment by vincentsarago** [comment]
⚠️ ⚠️ ⚠️ ⚠️  BETA Release ⚠️ ⚠️ ⚠️ ⚠️ 

Edit: I thought that doing a beta was safer but it seems that pining rio-tiler~=2.0a will still install 2.0b1, I'm really sorry for the trouble. 

## 2.0b1 - 2020/07/22
https://github.com/cogeotiff/rio-tiler/blob/72ddff276388f89915000930c4d77c8d45ca9eac/CHANGES.txt#L1-L11

This release brings a lot of changes ☝️, please checkout https://github.com/cogeotiff/rio-tiler#cogreader to see how to use the new COGReader.

**Comment by vincentsarago** [comment]
### 2.0b2 - 2020/07/23

https://github.com/cogeotiff/rio-tiler/blob/94e55538c5ff8e8e70ffd38b8011f471ec870d1c/CHANGES.txt#L1-L3

**Comment by vincentsarago** [comment]
### Road to 2.0 

- [ ] Add Notebooks #76 
- [ ] Update docs to mkdocs (@kylebarron)

**Comment by vincentsarago** [comment]
### 2.0b3 - 2020/07/27

https://github.com/cogeotiff/rio-tiler/blob/aa58b04baf1726cad49f743f79f8c7adf995c758/CHANGES.txt#L1-L3

Because rio-tiler-mosaic was a no requirements plugin we moved it here ;-) 

thanks @kylebarron and @geospatial-jeff

**Comment by vincentsarago** [comment]
### 2.0b4 - 2020/07/30

https://github.com/cogeotiff/rio-tiler/blob/ce156bc5aaa52c579ee0891998eef7a97e44469c/CHANGES.txt#L1-L9

**Comment by vincentsarago** [comment]
### 2.0b5 - 2020/07/31

https://github.com/cogeotiff/rio-tiler/blob/5da61965c198907fc92508214c871485609c22ae/CHANGES.txt#L1-L7

**Comment by vincentsarago** [comment]
### 2.0b6 - 2020/08/04

https://github.com/cogeotiff/rio-tiler/blob/065e9d6079e651a1488709a0df6db849374f39c6/CHANGES.txt#L1-L7

**Comment by vincentsarago** [comment]
## 2.0.b7 - 2020/08/21

https://github.com/cogeotiff/rio-tiler/blob/f1c59e833c1b5a7c2caae9f07d43531519f63dd0/CHANGES.txt#L1-L12

Pretty important refactor of the BaseClass reader and introduction of the MultiBaseReader to support STAC and rio-tiler-pds readers. 

We also switched from dataclass to `attrs` which is more flexible and has some interesting features.

**Comment by vincentsarago** [comment]
## 2.0b8 - 2020/08/24

https://github.com/cogeotiff/rio-tiler/blob/588e512b1d3b5767b02aa391666677f91cb5533e/CHANGES.txt#L1-L3

**Comment by vincentsarago** [comment]
## 2.0b14 - 2020/10/02

We have made couple releases since 2.0b8 🤦 

```
## 2.0b14 (2020-10-02)

* Added back the Conctext Manager methods in `rio_tiler.io.base.BaseReader` but not as `@abc.abstractmethod` (ref: https://github.com/cogeotiff/rio-tiler/pull/273#discussion_r498937943)
* Move `rio_tiler_pds.reader.MultiBandReader` and `rio_tiler_pds.reader.GCPCOGReader` to rio-tiler (https://github.com/cogeotiff/rio-tiler/pull/273)

## 2.0b13 (2020-10-01)

* remove ContextManager requirement for `rio_tiler.io.base.BaseReader` and `rio_tiler.io.base.MultiBaseReader` base classes.
* move ContextManager properties definition to `__attrs_post_init__` method in `rio_tiler.io.STACReader` and `rio_tiler.io.COGReader` (ref: https://github.com/cogeotiff/rio-tiler-pds/issues/21)

## 2.0b12 (2020-09-28)

* Make sure Alpha band isn't considered as an internal mask by `utils.has_mask_band`

## 2.0b11 (2020-09-24)

* reduce verbosity in `rio_tiler.tasks.filter_tasks` exception logging (#266).

2.0b10 (2020-09-15)
------------------
- add `post_process` callback to `rio_tiler.render._read` and `rio_tiler.render.point` to apply
specific operation ouput arrays.

2.0b9 (2020-09-09)
------------------
- restore Mkdocs search bar (#255)
- Allow class (not just instance) to be passed to pixel_selection (#250)
- Add Binder link/badge to README (#254)
- Add mkdocs-jupyter to show notebooks in website (#253)
- Remove deprecated functions (#247)
- Export modules from top-level package (#246)
- Allow overwriting colormap with force=True (#249)
- Pin black version (#251)
- Add contributing.md (#242)
- Add mkdocs config (#240)
- Add `NPY` support in `rio_tiler.utils.render` to save tile in numpy binary format (#256)
- Remove bare `Exception` and add more detailed errors (#248)

```

**Comment by vincentsarago** [comment]
## 2.0b15 - 2020/10/05
https://github.com/cogeotiff/rio-tiler/blob/5cd3ab81a056c68da25d9dc78c966971e44032fb/CHANGES.md#20b15-2020-10-05

```
- Fix missing Exception catching when running task outside threads (ref: https://github.com/developmentseed/titiler/issues/130).
- add rio-tiler logger (https://github.com/cogeotiff/rio-tiler/issues/277).
```

**Comment by vincentsarago** [comment]
### 2.0.0b17 - 2020/10/13

https://github.com/cogeotiff/rio-tiler/blob/master/CHANGES.md#200b17-2020-10-13

* switch to morecantile for TMS definition (ref: https://github.com/cogeotiff/rio-tiler/issues/283)
* add tms options in Readers (breaking change if you create custom Reader from BaseReader)
* add tile_bounds vs bounds check in tile methods for MultiBands and MultiBase classes
* add tile_exists method in BaseReader (take tms in account)
* adapt zooms calculation in COGReader
* add `LastBandHigh` and `LastBandLow` pixel selection (ref: https://github.com/cogeotiff/rio-tiler/pull/270)

Deprecated function

- rio_tiler.reader.tile
- rio_tiler.utils.geotiff_options
- rio_tiler.utils.tile_exists
- rio_tiler.io.multi_*

⚠️ ⚠️ ⚠️ This is a major update (and hope will be the last one). This version could break some user code if you were creating custom Reader using `rio_tiler.io.BaseReader` which now requirement a tms option. 

FYI: I'm targetting end of october for rio-tiler 2.0.0 final

**Comment by vincentsarago** [comment]
## 2.0.0b18 (2020-10-22)

* surface dataset.nodata in COGReader.nodata property (https://github.com/cogeotiff/rio-tiler/pull/292)
* fix non-threaded tasks scheduler/filter (https://github.com/cogeotiff/rio-tiler/pull/291)

**Comment by vincentsarago** [comment]
👋 I've just pushed a new version of rio-tiler - 2.0.0rc4 

```
* add NPZ output format (https://github.com/cogeotiff/rio-tiler/issues/308)
* add pystac for STAC item reader (author @emmanuelmathot, https://github.com/cogeotiff/rio-tiler/issues/212)
* delete deprecated function: rio_tiler.reader.tile, rio_tiler.utils.tile_exits and rio_tiler.utils.geotiff_options
* deprecated rio_tiler.mercator submodule (https://github.com/cogeotiff/rio-tiler/issues/315)
* update morecantile version to 2.1, which has better tms.zoom_for_res definition.
```
ref: https://github.com/cogeotiff/rio-tiler/blob/master/CHANGES.md#200rc4-2020-12-18

this is likely the last RC before the final 2.0.0 😱 😅

**Comment by vincentsarago** [comment]
👋 version 2.0 is going to be released today. 

We've made some additional changes and did a full re-write of all python Docstrings to enable API documentation. I've also added new notebook examples to help with the new features.

Thanks everyone who help (commits, reviews, issues ...) 🙏
